doc/content/xcp-networkd/host-network-device-ordering.md

```diff
@@ -0,0 +1,181 @@
+there are hardware changes, the position of a network device in the initial
+order should remain the same regardless of how many times the host is rebooted.
+
+To achieve this, the initial order should be saved persistently on the host's
+local storage so it can be referenced in subsequent orderings. When performing
+another ordering after the initial order has been saved, the position of a
+currently unordered network device should be determined by finding its position
+in the last saved order. The MAC address of the network device is a reliable
+attribute for this purpose, as it is considered unique for each network device
+globally.
+
+Therefore, the network devices in the saved order should have their MAC
+addresses saved together, effectively mapping each position to a MAC address.
+When performing an ordering, the stable position can be found by searching the
+last saved order using the MAC address.
+
+Name labels of the network devices are not considered reliable enough to
+identify particular devices. For example, if the name labels are determined by
+the PCI address via systemd, and a firmware update changes the PCI addresses of
+the network devices, the name labels will also change.
+
+
+Replacement
+-----------
+
+However, what happens when the MAC address of an unordered network device cannot
+be found in the last saved order? There are two possible scenarios:
+
+1. It's a newly added network device since the last ordering.
+2. It's a new device that replaces an existing network device.
+
+Replacement is a supported scenario, as an administrator might replace a broken
+network device with a new one.
+
+This can be recognized by comparing the PCI address where the network device is
+located. Therefore, the PCI address of each network device should also be saved
+in the order. In this case, searching the PCI address in the order results in
+one of the following:
+
+1. Not found: This means the PCI address was not occupied during the last
+ordering, indicating a newly added network device.
+2. Found with a MAC address, but another device with this MAC address is still
+present in the system: This suggests that the PCI address of an existing
+network device (with the same MAC address) has changed since the last ordering.
+The current unordered network device is considered newly added.
+3. Found with a MAC address, and no current devices have this MAC address: This
+indicates that a new network device has replaced the old one in the same PCI
+slot.
+
+The replacing network device should be assigned the same position as the
+replaced one.
+
+Removed devices
+---------------
+
+A network device can be removed or unplugged since the last ordering. Its
+position, MAC address, and PCI address are saved for future reference, and its
+position will be reserved. This means there may be a gap in the order: a
+position that was previously assigned to a network device is now vacant because
+the device has been removed.
```

Comment:
I believe we need to discuss when such a slot can be re-used. But I agree that the information from a removed card should be remembered.

Reply:
> when the reserved position is released and assigned to another device.

`replacement` and `removed and then added back` are just for this. Note the `replacement` can be `removed and then add a new card on the same PCI slot`.

Reply:
`replacement` is not `newly added`. The reserved position will not be released.